src/main/java/info/freeit/boarderpicker/service/Impl/SaleServiceImpl.java

```diff
@@ -1,58 +1,69 @@
 import info.freeit.boarderpicker.repository.GameRepository;
 import info.freeit.boarderpicker.repository.SaleRepository;
 import info.freeit.boarderpicker.repository.UserRepository;
 import info.freeit.boarderpicker.service.SaleService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 @Service
+@Transactional
 public class SaleServiceImpl implements SaleService {
 
-    @Autowired
-    private SaleRepository saleRepository;
-    @Autowired
-    private UserRepository userRepository;
-    @Autowired
-    private GameRepository gameRepository;
+    private final SaleRepository saleRepository;
+    private final UserRepository userRepository;
+    private final GameRepository gameRepository;
+
+    public SaleServiceImpl(SaleRepository saleRepository, UserRepository userRepository, GameRepository gameRepository) {
+        this.saleRepository = saleRepository;
+        this.userRepository = userRepository;
+        this.gameRepository = gameRepository;
+    }
 
     @Override
-    public List<Sale> getAllSales() {
-        return saleRepository.findAll();
+    public List<SaleDto> getAllSales() {
+        return saleRepository.findAll().stream().map(SaleDto::fromSale).toList();
     }
 
     @Override
-    public List<Sale> getSalesByUser(int userID) {
-        return saleRepository.findAll().stream().filter(sale -> sale.getUser().getId() == userID).toList();
+    public List<SaleDto> getSalesByUser(int userID) {
+        return saleRepository.findAll().stream().filter(sale -> sale.getUser().getId() == userID)
+                .map(SaleDto::fromSale).toList();
     }
 
     @Override
-    public List<Sale> getSalesByGame(int gameID) {
-        return saleRepository.findAll().stream().filter(sale -> sale.getGame().getId() == gameID).toList();
+    public List<SaleDto> getSalesByGame(int gameID) {
+        return saleRepository.findAll().stream().filter(sale -> sale.getGame().getId() == gameID).
+                map(SaleDto::fromSale).toList();
     }
 
     @Override
-    public Sale getSaleByID(int id) throws ObjectPersistenceException {
-        return saleRepository.findById(id)
-                .orElseThrow(() -> new ObjectPersistenceException(String.format("Sale with id %d is not found", id)));
+    public SaleDto getSaleByID(int id) {
+        Sale sale = saleRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException(String.format("Sale with id %d is not found", id)));
+        return SaleDto.fromSale(sale);
     }
 
     @Override
-    public Sale saveSale(Sale sale, int userID, int gameID) {
+    public SaleDto saveSale(UpdateSaleDto sale, BPUserDetails user, int gameID) {
         try {
-            User user = userRepository.findById(userID).get();
+            User userFromDB = userRepository.findById(user.getId()).get();
```

Comment:
User could be null, so getOrThrow(() -> ...); ? Maybe it'll be better

Reply:
Done. Many thanks for recommendation